NEWS.rst

```diff
@@ -9,6 +9,34 @@
 
 .. towncrier release notes start
 
+24.1b2 (2024-06-11)
+===================
+
+Features
+--------
+
+- Remove vendored colorama. (`#12702 <https://github.com/pypa/pip/issues/12702>`_)
+- Report informative messages about invalid requirements. (`#12713 <https://github.com/pypa/pip/issues/12713>`_)
+
+Bug Fixes
+---------
+
+- Eagerly import the self version check logic to avoid crashes while upgrading
+  or downgrading pip at the same time. (`#12675 <https://github.com/pypa/pip/issues/12675>`_)
+- Accommodate for mismatches between different sources of truth for extra names, for packages generated by ``setuptools``. (`#12688 <https://github.com/pypa/pip/issues/12688>`_)
+
+Vendored Libraries
+------------------
+
+- Remove vendored charset_normalizer.
+
+  Requests provides optional character detection support on some APIs
+  when processing ambiguous bytes. This isn't relevant for pip to function
+  and we're able to remove it due to recent upstream changes.
+- Upgrade packaging to 24.1
```

Comment:
I probably should've added a news fragment that this upgrade of packaging fixes https://github.com/pypa/pip/issues/12691... 

```md
- Upgrade packaging to 24.1 to fix dependency resolution crashes while running on development versions of CPython (#12691)
```